app/models/user.rb

```diff
@@ -1168,12 +1168,16 @@ def self.trainee_delegate_ids
   end
 
   def self.search(query, params: {})
-    users = Person.includes(:user).current
-    # We can't search by email on the 'Person' table
-    search_by_email = false
-    unless ActiveRecord::Type::Boolean.new.cast(params[:persons_table])
+    search_by_email = ActiveRecord::Type::Boolean.new.cast(params[:email])
+    admin_search = ActiveRecord::Type::Boolean.new.cast(params[:adminSearch])
+
+    return User.where(email: query) if admin_search && search_by_email
```

Comment:
Why is it necessary to return early if you're (admin) searching via email? What prevents admins from searching by WCA ID and email at the same time, for example?

Reply:
Irrespective of whether search by email or not, I'm introducing here `admin_search` which is supposed to be optimized. In this case, I'm searching by email, and I just want to return the actual value. Later, if the admin is searching by WCA ID, I can implement something which returns only one result and hence reduce unwanted querying in the whole DB.